composer_triggering_dags_with_functions_and_pubsub/main.tf

```diff
@@ -0,0 +1,191 @@
+# Copyright 2023 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START triggering_dags_with_functions_and_pubsub]
+#   Triggering DAGs Using Cloud Function and Pub/Sub Messages with Terraform
+#
+#   Usage:
+#       1. Select or create a new project that you will use to create the resources
+#       2. Replace <PROJECT_ID> with the project ID of your project
+#       3. Replace <PROJECT_NUMBER> with the account number of your project
+#       4. Upload into your directory the pubsub_function.zip file: https://github.com/akvelon/python-docs-samples/tree/main/composer_tutorials/triggering-dags-with-functions-and-pubsub/cloud_function
+#       5. Upload into your directory the DAG source code file: https://github.com/akvelon/python-docs-samples/tree/main/composer_tutorials/triggering-dags-with-functions-and-pubsub/DAG
+#       6. Set up your environment and apply the configuration using basic Terraform commands: https://cloud.google.com/docs/terraform/basic-commands
+#
+#   The script provisions the following resources in the project:
+#  	- Creates a VPC network and a subnetwork
+#	- Creates a Cloud Composer environment
+#	- Creates a Composer Service Agent account
+#	- Grants the Cloud Composer v2 API Service Agent Extension role and the Composer Worker role to the Composer Service Agent account
+#	- Creates a new Cloud Storage bucket
+#	- Creates a new Cloud Function and deploys the function source code from the pubsub_function.zip file to the Cloud Storage bucket
+#	- Uploads the example DAG source code from the specified file into the Cloud Composer bucket
+
+#
+
+###############################
+#                             #
+# Create Composer environment #
+#                             #
+###############################
+resource "google_composer_environment" "new_composer_env" {
+  name    = "composer-environment"
+  region  = "us-central1"
+  project = "<PROJECT_ID>"
```

Comment:
You can remove the `project` attribute from each resource. The samples in this repo assume that users setting the project ID elsewhere or are running Terraform in Cloud Shell and therefore the project ID is inherited from whatever project they're in.

Reply:
Or you can do this:

```
data "google_project" "project" {
}

resource .... {
  project = data.google_project.project.project_id
  ...
}
```